torch/csrc/Exceptions.h

```diff
@@ -14,7 +14,7 @@
   } catch (python_error &e) {                                                  \
     return retval;                                                             \
   } catch (std::exception &e) {                                                \
-    PyErr_SetString(PyExc_RuntimeError, e.what());                             \
+    PyErr_SetString(PyExc_RuntimeError, processErrorMsg(e.what()));            \
```

Comment:
Note that both of the following would be safe if processErrorMsg returns a `std::string`:

```c++
auto msg = processErrorMsg(e.what());
PyErr_SetString(PyExc_RuntimeError, msg.c_str());
```

This is safe because PyErr_SetString makes a copy of the `const char *`


```c++
PyErr_SetString(PyExc_RuntimeError, processErrorMsg(e.what()).c_str());
```

This is still safe because temporaries (the return value of processErrorMsg) last the lifetime of the **full-expression** that contains the point where they're created.

Reply:
Thanks for the explanation!